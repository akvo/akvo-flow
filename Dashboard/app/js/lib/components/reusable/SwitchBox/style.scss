.switch {
  position: relative;
  display: inline-block;
  width: 50px;
  height: 26px;

  input {
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + .slider {
      background-color: #424242;

      &::before {
        -webkit-transform: translateX(21px);
        -ms-transform: translateX(21px);
        transform: translateX(21px);
      }
    }
  }

  .slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ccc;
    -webkit-transition: 0.4s;
    transition: 0.4s;
    border-radius: 26px;

    &::before {
      position: absolute;
      content: '';
      height: 24px;
      width: 26px;
      left: 1px;
      bottom: 1px;
      background-color: white;
      -webkit-transition: 0.4s;
      transition: 0.4s;
      border-radius: 23px;
    }
  }
}
