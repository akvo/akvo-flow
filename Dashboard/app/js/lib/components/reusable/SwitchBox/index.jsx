import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

export default class SwitchBox extends React.Component {
  render() {
    const { id, name, checked } = this.props;

    return (
      <label htmlFor={id} className="switch">
        <input
          type="checkbox"
          id={id}
          name={name}
          checked={checked}
          onChange={() => this.props.onChange(name, !checked)}
          data-testid="switch-box"
        />
        <span className="slider round" />
      </label>
    );
  }
}

SwitchBox.defaultProps = {
  checked: false,
};

SwitchBox.propTypes = {
  checked: PropTypes.bool,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
