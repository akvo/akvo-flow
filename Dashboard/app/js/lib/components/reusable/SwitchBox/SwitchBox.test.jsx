import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import SwitchBox from './index';
import '@testing-library/jest-dom/extend-expect';

// https://github.com/testing-library/react-testing-library#suppressing-unnecessary-warnings-on-react-dom-168
// eslint-disable-next-line no-console
const originalError = console.error;
beforeAll(() => {
  // eslint-disable-next-line no-console
  console.error = (...args) => {
    if (/supports * the "act"/.test(args[0])) {
      return;
    }
    originalError.call(console, ...args);
  };
});

afterAll(() => {
  // eslint-disable-next-line no-console
  console.error = originalError;
});

// Main Tests
describe('SwitchBox Tests', () => {
  afterEach(cleanup);

  const props = {
    id: 'fake-id',
    name: 'fake-id',
    checked: false,
    onChange: jest.fn(),
  };

  it('+++ renders <snapshot>', () => {
    const wrapper = render(<SwitchBox {...props} />);

    expect(wrapper.container).toMatchSnapshot();
  });

  test('+++ SwitchBox works', () => {
    const wrapper = render(<SwitchBox {...props} />);

    // find input and trigger a check
    const inputNode = wrapper.getByTestId('switch-box');
    fireEvent.click(inputNode);

    expect(props.onChange).toHaveBeenCalledTimes(1);
    expect(props.onChange).toHaveBeenCalledWith('fake-id', true);
  });
});
