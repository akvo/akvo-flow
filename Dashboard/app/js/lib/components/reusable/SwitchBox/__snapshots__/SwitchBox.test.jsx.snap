// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchBox Tests +++ renders <snapshot> 1`] = `
<div>
  <label
    class="switch"
    for="fake-id"
  >
    <input
      data-testid="switch-box"
      id="fake-id"
      name="fake-id"
      type="checkbox"
    />
    <span
      class="slider round"
    />
  </label>
</div>
`;
